code/python/create_template_from_Syn_schema.py

```diff
@@ -0,0 +1,270 @@
+         - a blank template to use for data entry
+         - a dictionary defining the columns in the template
+         - a listing of allowable values for columns that use a controlled
+           vocabulary list
+
+Input parameters: Full name of the registered schema, including
+                      organization
+                  Full pathname to the output template file
+                  Desired output - either csv or excel
+
+Outputs: csv template file or Excel workbook
+
+Execution: create_template_from_Syn_schema.py <JSON schema name>
+             <output file> <csv/excel>
+"""
+
+import argparse
+import os
+import pandas as pd
+import synapseclient
+
+VALUES_LIST_KEYWORDS = ["anyOf", "enum"]
+
+def get_alias_dictionary(json_schema):
+    """
+    Function: get_alias_dictionary
+
+    Purpose: Given a dereferenced Synapse schema, return a dictionary of schema
+             names and their associated properties. This is necessary because
+             there are occasions where the property references a schema with
+             a different name. For example, in the PEC snpArray template
+             schema, the property is 260/280 but the schema it references is
+             qc260280 because Synapse does not allow non-alphanumeric
+             characters in the schema name and also requires that the schema
+             name starts with a character.
+
+    Input parameters: A dereferenced Synapse schema
+
+    Returns: A dictionary with the schema name as the key and the schema
+             property as the value.
+    """
+    alias_dict = {}
+    schema_properties = json_schema["validationSchema"]["properties"]
+
+    for key_val in schema_properties:
+        key = schema_properties[key_val]["$ref"].split(".")[-1]
+        alias_dict[key] = key_val
+
+    return alias_dict
+
+
+def get_Syn_definitions_values(json_schema, synlogin):
+    """
+    Function: get_Syn_definitions_values
+
+    Purpose: Return pandas dataframes of schema properties needed to generate
+             templates.
+
+    Input parameters: File object pointing to the JSON schema file
+                      Synapse object created in the main program to log into
+                      Synapse.
+
+    Returns: A dataframe of key types, definitions, required keys, and maximum
+             sizes
+                 definitions_df["key"] - string
+                 definitions_df["type"] - string
+                 definitions_df["description"] - string
+                 definitions_df["required"] - Boolean
+                 definitions_df["maximumSize"] - integer
+
+             A dataframe of key values lists
+                 values_df["key"] - string
+                 values_df["value"] - string
+                 values_df["valueDescription"] - string
+                 values_df["source"] - string
+
+    Note: This function is used with JSON schemas that are registered in
+          Synapse
+    """
+
+    import pandas as pd
```

Comment:
Unless I've missed something this inline import doesn't seem necessary since it is imported at the top of the file? Usually an inline import like this would be because you didn't want to enforce the strict dependency on an import which was only conditionally used or similar.

Reply:
Thanks for the catch!  The functions in this program had been in a separate module I'd created for manipulating schemas, and I pulled them in so that the program could stand alone.  Apparently I didn't clean them as well as I should have.